test/testdata/rbi/string.rb

```diff
@@ -24,3 +24,20 @@
 "foo".encode("encoding", "other_encoding", fallback: {})
 "foo".encode("encoding", fallback: {})
 "foo".encode(fallback: {})
+
+# match
+m1 = "foo".match("f")
+m2 = "foo".match("f", 1)
+T.assert_type(m1, T.nilable(MatchData))
```

Comment:
This (and all the other uses) should be `T.assert_type!`

Reply:
Thanks. I've updated it in 2523c5371.